src/commands.py

```diff
@@ -0,0 +1,189 @@
+        self.original_command = cmd_line
+        self.command_to_exe = ""
+        self.options = {
+            "-c": 1,
+            "--failed-count": float('inf'),
+            "--sys-trace": False,
+            "--call-trace": False,
+            "--log-trace": False,
+            "--debug": False,
+            "--help": False,
+            "--net-trace": False
+        }
+        self.return_codes = dict()
+        self.valid_command = True
+
+        # splits (according to whitespace) the cmd line into a list
+        args = cmd_line.split()
+
+        # sets the options and build the command
+        i = 0
+        while i < len(args):
+            if args[i] in self.options:
+                if args[i] == "-c" or args[i] == "--failed-count":
+                    i += 1
+                    if i < len(args) and args[i].isnumeric():
+                        self.options[args[i - 1]] = int(args[i])
+                    else:
+                        print("-c should have a zero or positive number after")
+                        self.valid_command = False
+                else:
+                    # set the flag
+                    self.options[args[i]] = True
+            # not one of our options - it's part of the command to execute
+            else:
+                self.command_to_exe += args[i] + " "
+            i += 1
+
+    def exe(self):
+        """Execute the command according to it's options.
+                    input:
+                        -self - The command.
+            """
+
+        if not self.valid_command:
+            return
+
+        count_failed = 0
+
+        # check options and act accordingly
+        for i in range(self.options.get("-c")):
+
+            # --help
+            if self.options.get("--help"):
+                self.help_message()
+            else:
+                # --call-trace
+                if self.options.get("--call-trace"):
+                    self.command_to_exe = "strace -c -o sys.txt " + self.command_to_exe
+
+                # run the command
+                return_value = os.system(self.command_to_exe + " > out.txt 2> err.txt")
+
+                # updates return codes of needed
+                self.return_codes[return_value] = self.return_codes[
+                                                      return_value] + 1 if return_value in self.return_codes else 1
+                # --debug
+                if self.options.get("--debug"):
+                    self.debug()
+
+                # in case the run failed, check options and act accordingly
+                if return_value != 0:
+                    print("An exception occurred")
+                    count_failed += 1
+
+                    # --sys-trace
+                    if self.options.get("--sys-trace"):
+                        self.sys_trace()
+
+                    # --call-trace
+                    if self.options.get("--call-trace"):
+                        sys_file = open("sys.txt", "r")
```

Comment:
Opening and closing files should be done in a try/finally block in order to avoid resource leak. In Python the `open` method can be used as a context manager to overcome this issue.